src/sidebar.cpp

```diff
@@ -0,0 +1,573 @@
+{
+    if( lvl > BODYTEMP_SCORCHING ) {
+        return 7;
+    } else if( lvl > BODYTEMP_VERY_HOT ) {
+        return 6;
+    } else if( lvl > BODYTEMP_HOT ) {
+        return 5;
+    } else if( lvl > BODYTEMP_COLD ) {
+        return 4;
+    } else if( lvl > BODYTEMP_VERY_COLD ) {
+        return 3;
+    } else if( lvl > BODYTEMP_FREEZING ) {
+        return 2;
+    }
+    return 1;
+}
+
+void player::disp_status( const catacurses::window &w, const catacurses::window &w2 )
+{
+    bool sideStyle = use_narrow_sidebar();
+    const catacurses::window &weapwin = sideStyle ? w2 : w;
+
+    {
+        const int y = sideStyle ? 1 : 0;
+        const int wn = getmaxx( weapwin );
+        trim_and_print( weapwin, y, 0, wn, c_light_gray, print_gun_mode( *this ) );
+    }
+
+    // Print currently used style or weapon mode.
+    std::string style;
+    const auto style_color = is_armed() ? c_red : c_blue;
+    const auto &cur_style = style_selected.obj();
+    if( cur_style.force_unarmed || cur_style.weapon_valid( weapon ) ) {
+        style = cur_style.name;
+    } else if( is_armed() ) {
+        style = _( "Normal" );
+    } else {
+        style = _( "No Style" );
+    }
+
+    if( !style.empty() ) {
+        int x = sideStyle ? ( getmaxx( weapwin ) - 13 ) : 0;
+        mvwprintz( weapwin, 1, x, style_color, style.c_str() );
+    }
+
+    wmove( w, sideStyle ? 1 : 2, 0 );
+    if( get_hunger() > 2800 ) {
+        wprintz( w, c_red,    _( "Starving!" ) );
+    } else if( get_hunger() > 1400 ) {
+        wprintz( w, c_light_red,  _( "Near starving" ) );
+    } else if( get_hunger() > 300 ) {
+        wprintz( w, c_light_red,  _( "Famished" ) );
+    } else if( get_hunger() > 100 ) {
+        wprintz( w, c_yellow, _( "Very hungry" ) );
+    } else if( get_hunger() > 40 ) {
+        wprintz( w, c_yellow, _( "Hungry" ) );
+    } else if( get_hunger() < -60 ) {
+        wprintz( w, c_green,  _( "Engorged" ) );
+    } else if( get_hunger() < -20 ) {
+        wprintz( w, c_green,  _( "Sated" ) );
+    } else if( get_hunger() < 0 ) {
+        wprintz( w, c_green,  _( "Full" ) );
+    }
+
+    /// Find hottest/coldest bodypart
+    // Calculate the most extreme body temperatures
+    int current_bp_extreme = 0, conv_bp_extreme = 0;
+    for( int i = 0; i < num_bp ; i++ ) {
+        if( abs( temp_cur[i] - BODYTEMP_NORM ) > abs( temp_cur[current_bp_extreme] - BODYTEMP_NORM ) ) {
+            current_bp_extreme = i;
+        }
+        if( abs( temp_conv[i] - BODYTEMP_NORM ) > abs( temp_conv[conv_bp_extreme] - BODYTEMP_NORM ) ) {
+            conv_bp_extreme = i;
+        }
+    }
+
+    // Assign zones for comparisons
+    const int cur_zone = define_temp_level( temp_cur[current_bp_extreme] );
+    const int conv_zone = define_temp_level( temp_conv[conv_bp_extreme] );
+
+    // delta will be positive if temp_cur is rising
+    const int delta = conv_zone - cur_zone;
+    // Decide if temp_cur is rising or falling
+    const char *temp_message = "Error";
```

Comment:
Slight inconsistency: this variable is initialized but each code path below assigns a new value. The same construct is used below with the `morale_str` variable, but it is *not* initialized.